src/main.cpp

```diff
@@ -3295,6 +3295,16 @@ bool CBlock::ConnectBlock(CTxDB& txdb, CBlockIndex* pindex, bool fJustCheck, boo
             // Only reject superblock when it is new And when QuorumHash of Block != the Popular Quorum Hash:
             if (IsLockTimeWithinMinutes(GetBlockTime(),15)  && !fColdBoot)
             {
+                CBitcoinAddress address(bb.GRCAddress);
+                bool validaddressinblock = address.IsValid();
+                if (!validaddressinblock)
+                {
+                    return error("ConnectBlock[] : Superblock staked with invalid GRC address in block");
+                }
+                if (!IsNeuralNodeParticipant(bb.GRCAddress, nTime))
+                {
+                    return error("ConnectBlock[] : Superblock staked by ineligible neural node participant");
+                }
                 if (!VerifySuperblock(superblock,pindex->nHeight))
                 {
                     return error("ConnectBlock[] : Superblock avg mag below 10; SuperblockHash: %s, Consensus Hash: %s",
```

Comment:
ya can do. will include it on testnet as well thou for when we do get a superblock action going.